### ISSUE TITLE
Distance/Area tool

### ISSUE THREAD

**Issue by HelenwoodsJNCC** [jncc/web-mapper-core#107]
Hi @andyb-esdm, we are wondering how simple it would be to add a tool to the mappers allowing users to draw a line/polygon to measure a distance or area?

**Comment by andyb-esdm** [comment]
Hi @HelenwoodsJNCC 
There's a nice example of this in the OpenLayers documentation for the version we use (4.6.5). It should return accurate distances and areas by calculating these on a sphere. I don't think it would take very long to adapt this for the mapper.

https://openlayers.org/en/v4.6.5/examples/measure.html

**Comment by HelenwoodsJNCC** [comment]
@andyb-esdm, sounds good to me @JordanPinder are you happy for this to be actioned? I imagine the tool button could go on the bottom right tool bar near the zoom to coordinates button.

**Comment by HelenwoodsJNCC** [comment]
@andyb-esdm We are happy to proceed with this. Let us know if you have any questions or when you are ready to test something out.

**Comment by andyb-esdm** [comment]
@HelenwoodsJNCC okay - I'll fit it in in the next week.

**Comment by andyb-esdm** [comment]
@HelenwoodsJNCC thought I should update you on this so you know I've been working on it.
The measure tool launches from the toolbar and you can draw lines or polygons from it to measure distances and areas. So it's progressing okay.
Hit a bit of a snag, which is when you click on the map it gets feature info from the layers by default. So you get the side window pop up with feature info - which is a bit annoying! I'll need to figure out how to turn get feature info off while the measure tool is active.
I also need to do a bit more on styling.
Spent about 0.5d so far and it should be complete in less than a day total.

**Comment by andyb-esdm** [comment]
@HelenwoodsJNCC @JordanPinder 
This is implemented now and on the ESDM test site. There is a new button on the toolbar in the bottom right that activates the measure tool. The implementation is similar to the OpenLayers example except that activating the tool puts the map into measure mode (stopping get feature info for instance) and deactivating it puts the map back into 'normal' mode.
I'm not very happy with the icon (which is supposed to be a ruler but looks more like a comb!) but I'll email separately about that.

**Comment by HelenwoodsJNCC** [comment]
@andyb-esdm thanks for the update - I have just had a play around and think it looks great. My only two comments on styling are:
Would it be better if the measure tool popped up in around the same place (or just below) as the zoom to coordinate function? Also if we styled it to look a bit more like the coordinate function could we have a clear button to remove previously drawn areas rather than having to close and reopen the tool?
@JordanPinder any thoughts?

**Comment by andyb-esdm** [comment]
@HelenwoodsJNCC Yes. I think those would be easy enough to do. Length, area, clear in a row and the control the same width as zoom to coordinate.

**Comment by andyb-esdm** [comment]
@HelenwoodsJNCC I've moved the measure tool and adjusted the width (the full width of the zoom to coordinate was too wide). I've also implemented clear functionality. Deployed to test site.

**Comment by HelenwoodsJNCC** [comment]
@andyb-esdm - Looks good to me, and easier to use this way too. @JordanPinder is on leave at the moment but can check it once he is back and then we should be ready to deploy all the recent developments.

**Comment by JordanPinder** [comment]
@HelenwoodsJNCC @andyb-esdm apologies, this slipped off my radar. Just had a browse and that tools look nice and simple to use :) happy for them to be deployed.

**Comment by HelenwoodsJNCC** [comment]
@andyb-esdm can you let us know when you are able to do the deployment? Thanks

**Comment by andyb-esdm** [comment]
@HelenwoodsJNCC Just discussed with @SimonAnnetts about scheduling a deployment and work on #108 
We can fit this in next week (say, Thursday). However, I think we'll need to coordinate with @jonparsonsjncc for the deployment since there will be some database updates. 
We also need to consider how we finalise #94 and #96 but I can exclude them from the user interface for a deployment next week.

**Comment by jonparsonsjncc** [comment]
Hi Andy

Hope you are well

This is timely as we wanted to have a call with you to talk through the deployment process – as we are currently building our new “self-healing” geo server environment which will be created through a terraform script and we wanted to look at the deployment process in detail as part of this.

Would you have half an hour for a quick meeting with our dev team to quickly talk through the process – would you be free Friday between 2 and 3

Cheers

Jon


Jon Parsons
Technical Delivery Manager
(DIP DigM)
JNCC, Monkstone House, City Road, Peterborough, PE1 1JY
Tel: 01733 562626, Direct Dial: 01733 866827, Mobile: 07526 497077
Email: jon.parsons@jncc.gov.uk<mailto:jon.parsons@jncc.gov.uk>
Skype: jonparsons
My working days are: Monday to Friday
My pronouns are he/him

**Comment by andyb-esdm** [comment]
@jonparsonsjncc I've sent you an email about scheduling a meeting. Speak soon.
